examples/react/next-routing/pages/index.tsx

```diff
@@ -69,7 +70,13 @@ export default function HomePage({ serverState, url }: HomePageProps) {
           }),
         }}
         insights={true}
+        // only for tests
+        onStateChange={({ uiState, setUiState }) => {
+          setOnStateChangeCalled((times) => times + 1);
+          setUiState(uiState);
+        }}
       >
+        <output id="onStateChange">{onStateChangeCalled}</output>
```

Comment:
I think we probably should either have this in a test.tsx file or a separate example, what do you think?